react/src/index.tsx

```diff
@@ -4,40 +4,78 @@ import './index.css';
 import App from './app';
 import {mutators} from '../../shared/mutators';
 import {Replicache} from 'replicache';
-import {getPokeReceiver} from './poke';
-import {createSpace, spaceExists} from './space';
+import {assert} from './assert';
 
 const {pathname} = window.location;
 const paths = pathname.split('/');
-const spaceIDUrlParam =
+
+const spaceID =
   paths.indexOf('listID') > -1 ? paths[paths.indexOf('listID') + 1] : '';
-let spaceID = '';
-if (spaceIDUrlParam) {
-  spaceID = (await spaceExists(spaceIDUrlParam))
-    ? spaceIDUrlParam
-    : await createSpace(spaceIDUrlParam);
-} else {
-  spaceID = await createSpace();
+if (!spaceID || !(await spaceExists(spaceID))) {
+  window.location.href = '/listID/' + (await createSpace());
 }
 
-if (spaceIDUrlParam !== spaceID) {
-  window.location.href = '/listID/' + spaceID;
-}
+// See https://doc.replicache.dev/licensing for how to get a license key.
+const licenseKey = import.meta.env.VITE_REPLICACHE_LICENSE_KEY;
+assert(licenseKey, 'Missing VITE_REPLICACHE_LICENSE_KEY');
 
 const r = new Replicache({
-  // See https://doc.replicache.dev/licensing for how to get a license key.
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  licenseKey: import.meta.env.VITE_REPLICACHE_LICENSE_KEY!,
+  licenseKey,
   pushURL: `/api/replicache/push?spaceID=${spaceID}`,
   pullURL: `/api/replicache/pull?spaceID=${spaceID}`,
   name: spaceID,
   mutators,
 });
 
-const pokeReceiver = getPokeReceiver();
-pokeReceiver(spaceID, async () => {
-  await r.pull();
+// Implements a Replicache poke using Server-Sent Events.
+// If a "poke" message is received, it will pull from the server.
+const ev = new EventSource(`/api/replicache/poke-sse?spaceID=${spaceID}`, {
```

Comment:
`poke-sse` is an artifact of when we had multiple ways to poke in the next example. In this example, there is only one so the distinction is mental overhead for no reason. Please remove it. There should be no reference to `sse` in the code.

Reply:
required upgrade of express server but this is corrected now